### DISCUSSION TITLE
Can I use NEB images and/or geo opt results to train NequIP and/or Allegro parameters?

### DISCUSSION THREAD

**Discussion by turbosonics** [mir-group/nequip#355] [Q&A]
Hello. 

I'm learning this tool by following tutorials. But just in case I'm asking, can I include NEB images to the training set of NequIP or Allegro? 

For example, let's assume I have some AIMD results of crystal structure of interest from VASP or QE. And then I operated five NEB searches for specific cation-vacancy long range diffusion, generated 20 images for five diffusion pathways. That is total 100 images for 5 pathways, and each image have xyz coordinates of atoms, energies, and forces. 

I hope my NNFF to describe this cation kinetics and diffusion really nicely, so I hope to combine AIMD results and NEB results in the same training set for the NequIP parameters of this crystal. Is this possible in NequIP and/or Allegro? 

If not, what would be DFT suggestion to get the diffusion pathway correctly for NequIP/Allegro training? 

Similarly, if I conduct some bond distance scan or bond angle scan (multiple constrained geometry optimizations) from DFT, can I include them in the training set with AIMD results for NequIP or Allegro training? I don't know if NequIP and Allegro could read results from molecular orbital DFT tools like Gaussian. But I think I can try some "scanning" geo opts to get bond dissociation energy curve using VASP or QE. 

Thank you for your answer in advance. 

Best,

**Comment by Linux-cpp-lisp** [comment]
Hi @turbosonics ,

Thanks for your interest in our code!

Yes, Allegro & NequIP are entirely agnostic to the source of the training data, which can be a pile of individual frames that are sampled in any way you can think of (NEB, MD, Monte Carlo, something else, any combination thereof).  Generally, if you've already run the calculations, there is no reason not to include the extra data.  (The exception to this, of course, is if it would mean mixing calculations from different levels of theory / DFT settings, which is something you should only ever do _very_ intentionally and with a good reason.)

In general the key is to make sure that your training set samples a sufficiently diverse set of conformations around the areas of state space you care about doing inference on.  For example, if we were using MD to sample a training set that we want to use to run MD, often we will run the training MD at a higher temperature to ensure it comprehensively samples the states (even the rare ones) accessible to the inference simulation.

Regarding constrained optimizations, absolutely!  And if you have the data for intermediate frames from the optimizations (not just end frames) feel free to include those too, they can only help.
